### PR TITLE
hide the cursor after inactivity

### PR DESCRIPTION
**Pull request by l-j-g**
not sure how other people feel about this change, but it is an improvement imo.

after 5 seconds of inactivity the mouse cursor is hidden

### CODE REVIEW

**Comment by boredland** [comment]
While I like it, we probably should make it configurable.

File: community/sway/etc/sway/definitions [hunk 1]
```diff
@@ -95,6 +95,9 @@ set $swayidle swayidle -w \
     before-sleep 'playerctl pause' \
     before-sleep 'exec $locking & sleep $sleep_delay'
 
+# hide cursor after 5 seconds of inactivty 
+seat seat0 hide_cursor 5000
```
Comment (boredland) [inline community/sway/etc/sway/definitions]:
```suggestion
# hide cursor while typing
seat * hide_cursor when-typing enable
```

**Review by boredland — CHANGES_REQUESTED** [review]
wdyt about just hiding it while typing automatically?

**Comment by l-j-g** [comment]
to make it configurable we run into the same issue as the parameters for the sway-idle settings from some months ago. the system configuration is applied first, then the user configuration on top, which means the entire setting would need to be redefined. 

ideally we would set configurable parameters like that at as environment variables during the initialization of the shell (.config/profile/) and reference them within the sway configuration (is that possible ?, it should be right ?)

i think when typing might be a suitable 'no-config' compromise, ill try it out tonight.

**Comment by boredland** [comment]
I have been using it with the current state for some days now and quite like it. If users like to opt out, they can override the setting quite easily, so I don't see an issue. And I don't assume users will be annoyed by it anyway, as if you do something mouse-intensive on your machine, it probably will never even hide anyway.

**Comment by fourpoints** [comment]
This change broke a few things for me, there are cases where I'm holding down the mouse button and press a keyboard button for an action. It was tricky to notice, since the cursor was already hidden during dragging.

Related issue: https://github.com/swaywm/sway/issues/6297

I don't know if it's worth reverting, as I can see it being nice, but I thought I'd mention it in case anyone else has a similar issue.

**Comment by boredland** [comment]
Ya, for me copy-ing something from a type-hint in vscode intellisense broke - which is annoying too. I guess we should revert it.

**Comment by boredland** [comment]
instead of reverting we probably should go back to @l-j-g initial proposal of doing it after 5s of inactivity?

**Comment by fourpoints** [comment]
I've tried the 5 second hide for a few days now without issues, so it seems fine.

**Pull request merged**